app/utils/gitlab_manager.py

```
@@ -58,12 +58,28 @@ def get_repos(self, page=1, per_page=20):
+    def authenticate(self, access_token):
+        try:
+            gl = gitlab.Gitlab(url=self.base_url, private_token=access_token)
+            gl.auth()
+            return AuthenticatedGitLabManager(base_url=self.base_url, git_client=gl, access_token=access_token)
+
+        except Exception as e:
+            raise DevDoxAPIException(
+                user_message="GitLab authentication failed",
+                log_message="Failed to authenticate GitLabManager",
+                root_exception=e
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)